### ISSUE TITLE
Can you add Instagram share as well

### ISSUE THREAD

**Issue by satishnvrn** [nygardk/react-share#146]
Please add Instagram share as I'm using this plugin for pretty much every share. It would be very helpful if I get instagram share as well from this plugin

**Comment by Dennisboyangzhang** [comment]
Same situations for me. Greatly appreciated if Instagram was added!

**Comment by mikhailbalin** [comment]
There is no Instagram support for sharing. How do you do that?

**Comment by ReasonableDeveloper** [comment]
If there is no share button, could we have an `InstagramIcon`?

**Comment by acoulon99** [comment]
Is there a way to deep link into the Instagram mobile app (if installed), and pre-fill in an image, comment, and hashtags?

**Comment by i-schuetz** [comment]
Yes please! I use this library too for social media links (to simply open the social media account) and it would be handy if it included Instagram.

**Comment by nicholasglazer** [comment]
Now I need to install font to add this instagram icon. It would be a great thing just to have at least an icon in this library.

**Comment by ihaseebkhan** [comment]
Please add instagram support :3

**Comment by jaymathew** [comment]
I am in the same boat, would love to see instagram support added to this library.

**Comment by RahulJanagouda** [comment]
Please add support for instagram

**Comment by fabianstarke** [comment]
You definitively need to add instagram to the package ! Great library, thanks for this

**Comment by viveknimkarde** [comment]
+1

**Comment by RahulJanagouda** [comment]
@nygardk may be this could help https://developers.facebook.com/docs/instagram-api/guides/content-publishing

**Comment by JeffStodd** [comment]
Are there any plans to add this feature?

**Comment by BATTLEROYALXS** [comment]
Thank you for the lib dude ! 
Looking for instagram too

**Comment by marlonwilliams** [comment]
guys, why can't you add an Instagram icon lol? users have been asking for it for like 2 years now

**Comment by r-i-c-k** [comment]
lack of instagram is a big gap in this otherwise comprehensive library

**Comment by waqasshaukat** [comment]
instagram???

**Comment by dcworldwide** [comment]
If you need it, submit a PR.

**Comment by NemanjaMandic** [comment]
Still no icon for Instagram, for years. Really!?

**Comment by kmjennison** [comment]
Please be more considerate to open source maintainers, who are often doing this work for free in their spare time. The "+1" spam doesn't help. If this feature matters to you, you can always contribute a PR or fork the repo.

**Comment by muhammedorhungunes** [comment]
Oppsss!! please add to instagram icon from react-share

**Comment by BasirBaidya** [comment]
Hi Team, We definitely need **Instagram** share as well. Please add this any how it possible. Thank you

**Comment by Johnrobmiller** [comment]
> Please be more considerate to open source maintainers, who are often doing this work for free in their spare time. The "+1" spam doesn't help. If this feature matters to you, you can always contribute a PR or fork the repo.

I agree that people should be considerate in general, but I personally don't see any evidence the people are being inconsiderate here.  With that said, expressing frustration about a genuine flaw that this product has is not inconsiderate.  Instead, it is normal and expected.

**Comment by huzaifa-99** [comment]
I would like to make a PR on it. If there is no one currently doing that?

**Comment by aminnoura** [comment]
Does Instagram added or not yet?

**Comment by Rushil-2311** [comment]
instagram added or not ???

**Comment by GitanjaliCS** [comment]
Can you please add Instagram share button? I have tried to share on Instagram by manual coding but no success. Please I need this share button. 

Waiting for your reply.

Thank you in advance.

**Comment by sthadevraj** [comment]
How to share firebase image and other text.I am trying to share but only link is shown but not image.please reply me. I  would be vary thankful if this issue is solved.Thanks in advance.

**Comment by ijhar88** [comment]
anyone got instagram working.?

**Comment by mosesoak** [comment]
Hey I'm not a maintainer of this lib but thought someone should just say it clearly:

**Instagram doesn't currently support direct sharing for regular folks!** Sorry!

It's a choice by their team. 

What we plan to do for that use case is to pre-create specific Insta posts related to the content, then direct link to that post -- once a user lands there they can choose to add it to their insta story. This is a half-decent solution but requires that unique insta urls are included in the CMS for each piece of content.

So, in theory a simple Insta icon button might be useful for people here, but it would break the unified API of this library which always includes the url to share -- it would only be able to take an instagram url as an input.

**Comment by vdavidmarques** [comment]
is there news about instagram?

**Comment by NingXinHui** [comment]
Hope to share on instagram

**Comment by Sebastian2908-2007** [comment]
2023 no insta share still? It must be really annoying to do?

**Comment by yoganathanistan** [comment]
:( NO INSTAGRAM !

**Comment by SaraKhurshid28** [comment]
So which way use for Instagram share button

**Comment by tttp** [comment]
the [instagram documentation](https://developers.facebook.com/docs/instagram-api/guides/content-publishing) referenced is requiring the [User access token](https://developers.facebook.com/docs/facebook-login/access-tokens/#usertokens).

Unless I am missing something, it means that the user needs to approve the app first before being able to create an image/share, beside being an horrible workflow for the user, it seems to be clearly outside the score of a "simple" share component.

do you have an example anywhere of a working share on instagram that works? could you paste the link?

**Comment by tttp** [comment]
@guidodinatale you mean instagram or spotify?

**Comment by guidodinatale** [comment]
@tttp Spotify. 
It works through the app or if you are using the web app from a phone.

**Comment by ravjotsandhu** [comment]
6 years, still no instagram support?

**Comment by haashirnawaz9** [comment]
just use a link property from next/link and wrap your instagramicon and use href.

**Comment by avdheshpatel-scout** [comment]
Please add Instagram share as well.

**Comment by DevKaranMehta** [comment]
Please add Instagram share as well.
